### PR TITLE
[WIP] Make SpatialWorkerConnection multithreaded

### PR DESCRIPTION
**Pull request by Vatyx**

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp [hunk 1]
```diff
@@ -472,3 +485,180 @@ void USpatialWorkerConnection::OnConnectionFailure()
...
+bool USpatialWorkerConnection::Init()
+{
+	return true;
+}
+
+uint32 USpatialWorkerConnection::Run()
+{
+	while (KeepRunning)
+	{
+		FPlatformProcess::Sleep(0.015f);
```
Comment (girayimprobable) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp]: This is not sleeping long enough imo.

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp [hunk 2]
```diff
@@ -472,3 +485,180 @@ void USpatialWorkerConnection::OnConnectionFailure()
...
+}
+
+void USpatialWorkerConnection::Stop()
+{
+	KeepRunning.AtomicSet(false);
+}
+
+void USpatialWorkerConnection::InitializeWorkerThread()
+{
+	Thread = FRunnableThread::Create(this, TEXT("SpatialWorkerConnectionWorker"), 0, TPri_AboveNormal);
```
Comment (girayimprobable) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp]: Why `TPri_AboveNormal`? No need, unless you think it's justified?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp (hunk 1 above)
Comment (Vatyx) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp]: Totally. Just was there for testing. Was going to also bring up, should this be configurable?

Comment (girayimprobable) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp]: No doesn't need to be configurable at this stage unless there is explicit need from a customer. Make it 30hz?

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp]: Discussed offline, let's make this configurable.

**Comment by Vatyx** [comment]
Closing to open it up on a different branch.

**Pull request closed without merging**